tests/infrastructure/conftest.py

```diff
@@ -1,25 +1,100 @@
-import bitmath
-import pytest
-
+import logging
+from pathlib import Path
 
-@pytest.fixture(scope="session")
-def hugepages_gib_values(workers):
-    """Return the list of hugepage sizes (in GiB) across all worker nodes."""
-    return [
-        int(bitmath.parse_string_unsafe(value).GiB)
-        for worker in workers
-        if (value := worker.instance.status.allocatable.get("hugepages-1Gi"))
-    ]
+import pytest
+from kubernetes.dynamic.exceptions import ResourceNotFoundError
+from timeout_sampler import TimeoutExpiredError
 
+from tests.infrastructure.utils import (
+    verify_numa_enabled,
+    verify_tekton_operator_installed,
+)
+from tests.utils import verify_cpumanager_workers, verify_hugepages_1gi, verify_rwx_default_storage
+from utilities.exceptions import ResourceMissingFieldError, ResourceValueError
+from utilities.pytest_utils import exit_pytest_execution
 
-@pytest.fixture(scope="session")
-def xfail_if_no_huge_pages(hugepages_gib_values):
-    """Mark tests as xfail if the cluster lacks 1Gi hugepages."""
-    if not hugepages_gib_values or max(hugepages_gib_values) < 1:
-        pytest.xfail("Requires at least 1Gi hugepages on some node")
+LOGGER = logging.getLogger(__name__)
 
 
 @pytest.fixture(scope="session")
-def hugepages_gib_max(xfail_if_no_huge_pages, hugepages_gib_values):
+def hugepages_gib_max(hugepages_gib_values):
     """Return the maximum 1Gi hugepage size, capped at 64Gi."""
+    if not hugepages_gib_values:
+        raise ResourceValueError("Cluster does not report any 1Gi hugepages")
     return min(max(hugepages_gib_values), 64)
+
+
+@pytest.fixture(scope="session", autouse=True)
+def infrastructure_special_infra_sanity(
+    request,
+    admin_client,
+    junitxml_plugin,
+    schedulable_nodes,
+    hugepages_gib_values,
+):
+    """
+    Validates infrastructure requirements based on test markers.
+    """
+    skip_infra_sanity_check = "--skip-infra-sanity-check"
+
+    if request.session.config.getoption(skip_infra_sanity_check):
+        LOGGER.info(f"Sanity checks skipped because {skip_infra_sanity_check} was provided")
+        return
+
+    # Collect markers from infrastructure tests
+    infra_root = Path(request.config.rootpath) / "tests" / "infrastructure"
+    collected_marker_names = set()
+
+    for item in request.session.items:
+        if item.path.is_relative_to(infra_root):
+            collected_marker_names.update(marker.name for marker in item.iter_markers())
+    LOGGER.info(f"Collected markers from infrastructure tests: '{collected_marker_names}'")
+
+    if not collected_marker_names:
+        LOGGER.info("No infrastructure markers found, skipping verification checks")
+        return
```

Comment:
is there any chance this code will be reached? i.e the fixture is called when tests/infrastructure is collected by pytest